main_test.go

```diff
@@ -273,12 +274,19 @@ type MACFinRow struct {
 }
 
 type TestCase struct {
-	Name               string
-	PasswordManagerIn  []PasswordManagerRow
-	PasswordManagerOut []PasswordManagerRow
-	MACFinIn           []MACFinRow
-	UntrackedPasswords map[string]string // user -> password
-	ServerErrors       map[string]string // user -> path
+	Name                                    string
+	PasswordManagerIn                       []PasswordManagerRow
+	PasswordManagerOut                      []PasswordManagerRow
+	MACFinIn                                []MACFinRow
+	UntrackedPasswords                      map[string]string // user -> password
+	ServerErrors                            map[string]string // user -> path
+	InvalidMACFinInUsernameHeading          bool
```

Comment:
I like this solution to the question of how to specify a heading issue in the context of these test cases.

One thing I would suggest is, instead of having all these bools, have something like

```
type SheetProblem int
const (
  NoSheetProblem SheetProblem = iota
  SheetProblemInvalidMACFinUsernameHeading  # remove "In" from the names
  SheetProblemPasswordManagerEmpty
  // etc.
)

type TestCase struct {
  SheetInProblem SheetProblem
}

```

This makes it clearer that the test code does not support arbitrary combinations of problems, and also lets you write `tc.SheetProblem != NoSheetProblem` to see if there is a validation error expected, instead of the long disjunction of all the different booleans.

Reply:
Much better. Much cleaner. Much easier to understand.